lib/contacts.py

```diff
@@ -54,15 +55,23 @@ def import_file(self, path):
         try:
             with open(path, 'r') as f:
                 d = self._validate(json.loads(f.read()))
-        except json.decoder.JSONDecodeError:
+        except ValueError:
             traceback.print_exc(file=sys.stderr)
-            raise FileImportFailedEncrypted()
+            raise FileImportFailedInvalidJSON()
```

Comment:
You are right, the file might not be encrypted, all we know is that JSON decoding failed. `FileImportFailedEncrypted` was just a guess. However I don't think we should start introducing all kinds of  `File*Failed*` exceptions. 
How about keeping only `FileImportFailed` and `FileExportFailed`, and extending `FileImportFailed` with a message the same way you have done for `FileExportFailed`?

Reply:
ok will do. also shall i do this same treatment for invoices? invoices are payment requests right?

Reply:
yes they are the same as payment requests. i'll do invoice exporting as well.

Reply:
Invoices are what you should pay; i.e. the stuff at the bottom of the `Send` tab (when visible).